### PR TITLE
Refactor SBUS handling code (potential race condition fixed)

### PR DESCRIPTION
**Pull request by digitalentity**
Make S.Bus data updates atomic. Intoroduce better buffer handling (don't wait until timeout to start next frame reception). Could be a fix for https://github.com/iNavFlight/inav/issues/2981

### CODE REVIEW

File: src/main/rx/sbus_channels.c [hunk 1]
```diff
@@ -68,15 +68,14 @@ uint8_t sbusChannelsDecode(rxRuntimeConfig_t *rxRuntimeConfig, const sbusChannel
...
-    if (channels->flags & SBUS_FLAG_SIGNAL_LOSS) {
-    }
     if (channels->flags & SBUS_FLAG_FAILSAFE_ACTIVE) {
         // internal failsafe enabled and rx failsafe flag set
         // RX *should* still be sending valid channel data, so use it.
         return RX_FRAME_COMPLETE | RX_FRAME_FAILSAFE;
     }
-
-    return RX_FRAME_COMPLETE;
+    else {
```
Comment (fiam) [inline src/main/rx/sbus_channels.c]: I'd remove this else since you have a return in the preceding if.

**Review by fiam — APPROVED** [review]
Working fine on the bench. I don't have any objections to merge it as is, but it should be flight tested before it goes into a release.

**Comment by giacomo892** [comment]
Tested on the bench with a non FrSky receiver. Works.

**Comment by digitalentity** [comment]
I suppose we can merge it now. I've tested with multiple receivers, all seem to work just fine.

**Pull request merged**